### DISCUSSION TITLE
what's going on with the syntax of createStyles?

### DISCUSSION THREAD

**Discussion by Paraprax** [mantinedev/mantine#2800] [Q&A]
Forgive the crudity and ignorance of this question, but I've been using createStyles for months now by wrote and am still stumped.

AFAIK, when we call createStyles(), we pass it an unnamed callback, with no function body, and our styles object as its argument.

ie.
`createStyles((theme) => ({class1Name: class1Values, class2Name: class2values}) {this is where the function body would be, but is empty})`

my confusion is that, in the callback syntax I'm familiar with anywhere else, the argument for the anonymous callback function is already passed into the parameter upon declaration, before our arrow.

ie.
in a declaration like:
`createStyles((theme)  => {function body});`
'theme' would already be the parameter for the function body after the arrow. So, how can we have syntax like `createStyles((parameter) => (another parameter?){ function body});`

If I'm misinterpreting this syntax, or if my explanation of it here makes no sense, my core question is just: **how can we break down the syntax of createStyles(), piece by piece?**

The Mantine page for it does not use the words 'callback' or 'object' at any point, and what little I can find about it elsewhere online likewise glosses over the idiosyncrasy of its syntax. :neutral_face:

I can just keep using it by rote without error of course, but obv keen as ever to actually understand what's going on here rather than memorizing any individual quirks. No matching discussions found for this here yet. Thanks!

**Comment by rtivital** [comment]
You can pass parameters to createStyles – https://mantine.dev/styles/create-styles/#styles-parameters
If you are interested how the function is built you can explore the source code – https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/tss/create-styles.ts#L1
